### ISSUE TITLE
Support Yarn v3

### ISSUE THREAD

**Issue by BruceHaley** [dependabot/dependabot-core#5322]
We can no longer use dependabot because it does not support yarn v3 (or v2).

Do you have any plans to support v3?

Can you suggest a workaround?

Thank you.

**Comment by xfix** [comment]
This is a duplicate of #1297.

**Comment by bmulholland** [comment]
[Following instructions for Yarn's v2 upgrade](https://yarnpkg.com/getting-started/migration#step-by-step), which I did shortly after https://github.com/dependabot/dependabot-core/issues/1297 was released, actually upgrades to the latest Yarn version, not v2 (specifically `yarn set version berry`, which currently installs v3.2.4). Further, when committing that, dependabot doesn't throw an appropriate error or warning. Instead, it silently stops creating PRs because every dependabot run times out ("timeout running job"). (@jurre, FYI it's easy to accidentally create this situation, since it seems like something that's very easy to run into and I suspect will be reported by others soon. Maybe knowing this will save you some time investigating.)

Also, it turns out yarn v2 only supports [up to typescript 4.4](https://github.com/yarnpkg/berry/blob/cherry-pick/2.4/packages/plugin-compat/extra/typescript/gen-typescript-patch.js#L79)[1] when linking to node_modules (that is, no PnP[2]), so this means that, in this situation, dependabot does not support typescript > 4.4. Yarn v3 has more patches[1] and, [if the patch doesn't work, it at least doesn't block the install](https://github.com/yarnpkg/berry/issues/2935#issuecomment-911299992).

Personally, I only figured out I was on an unsupported Yarn v3 when I came to file an issue that Yarn v2 support was timing out dependabot. Now I'm trying to roll back to Yarn v2, and I'm having to downgrade Typescript to an older version, when the latest version of typescript is supported in both Yarn v1 and v3.

[1] Note that trunk yarn has patches for later versions; these just aren't backported, because (before this dependabot restriction) I imagine everyone could just upgrade to Yarn 3. https://github.com/yarnpkg/berry/blob/master/packages/plugin-compat/extra/typescript/gen-typescript-patch.js#L93
[2] And unfortunately, some tools, [like Nuxt](https://github.com/nuxt/framework/issues/5411), don't yet support PnP.

**Comment by bmulholland** [comment]
Actually, Yarn v1 seems generally much better supported than v2, where everyone has moved to v3. So the current dependabot support requires several sacrifices.

Beyond typescript, above, `yarn plugin import` breaks on Yarn v2:

```
% yarn plugin import interactive-tools
➤ YN0000: Downloading https://github.com/yarnpkg/berry/raw/master/packages/plugin-interactive-tools/bin/%40yarnpkg/plugin-interactive-tools.js
➤ YN0001: HTTPError: Response code 404 (Not Found)
    at se.<anonymous> (.../.yarn/releases/yarn-2.4.3.cjs:23:10082)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
➤ YN0000: Failed with errors in 0s 624ms
```

**Comment by xfix** [comment]
> [Following instructions for Yarn's v2 upgrade](https://yarnpkg.com/getting-started/migration#step-by-step), which I did shortly after #1297 was released, actually upgrades to the latest Yarn version, not v2 (specifically `yarn set version berry`, which currently installs v3.2.4). Further, when committing that, dependabot doesn't throw an appropriate error or warning. Instead, it silently stops creating PRs because every dependabot run times out ("timeout running job"). (@jurre, FYI it's easy to accidentally create this situation, and I suspect others will be reporting it soon. Maybe knowing this will save you some time investigating.)
> 
> Also, it turns out yarn v2 only supports [up to typescript 4.4](https://github.com/yarnpkg/berry/blob/cherry-pick/2.4/packages/plugin-compat/extra/typescript/gen-typescript-patch.js#L79)[1] when linking to node_modules (that is, no PnP[2]), so this means that, in this situation, dependabot does not support typescript > 4.4. Yarn v3 has more patches[1] and, [if the patch doesn't work, it at least doesn't block the install](https://github.com/yarnpkg/berry/issues/2935#issuecomment-911299992).
> 
> Personally, I only figured out I was on an unsupported Yarn v3 when I came to file an issue that Yarn v2 support was timing out dependabot. Now I'm trying to roll back to Yarn v2, and I'm having to downgrade Typescript to an older version. Frustratingly, the latest version of typescript is supported in both Yarn v1 and v3.
> 
> [1] Note that trunk yarn has patches for later versions; these just aren't backported, because (before this dependabot restriction) I imagine everyone could just upgrade to Yarn 3. https://github.com/yarnpkg/berry/blob/master/packages/plugin-compat/extra/typescript/gen-typescript-patch.js#L93 [2] And unfortunately, some tools, [like Nuxt](https://github.com/nuxt/framework/issues/5411), don't yet support PnP.

Dependabot support Yarn v3 (https://github.blog/changelog/2022-10-20-dependabot-can-now-generate-security-and-version-updates-for-yarn-v2-and-v3/), if you have any issues with it, it probably would be a good idea to make a separate issue for it.

**Comment by bmulholland** [comment]
Oh interesting, I thought it was only Yarn v2. Thanks!

**Comment by jeffwidman** [comment]
Closing as we now support Yarn v3, as mentioned ☝️.
